op-node/rollup/driver/state.go

```diff
@@ -323,6 +355,10 @@ func (s *Driver) eventLoop() {
 			delayedStepReq = nil
 			step()
 		case <-stepReqCh:
+			// Don't start the derivation pipeline until we are done with EL sync
+			if s.engineController.IsEngineSyncing() {
+				continue
+			}
```

Comment:
If we never receive an unsafe payload via p2p, and never trigger other engine operations through derivation, do we get back any Engine API response that can get us unstuck from a syncing engine state? We may want to repeat the last forkchoice call periodically just for the sync status response, to get unstuck upon sync completion. A simple solution might be to only `continue` if it's been less than X minutes since last derivation call, in the case the engine is syncing.

Reply:
We pretty heavily rely on the gossip network to be working in order to work with snap sync. If there's no gossipped blocks, we can't drive it in the first place.

Reply:
I'm thinking of the reverse case, when p2p stops (e.g. sequencer p2p gossip issue, or p2p issue on user end), and the engine is still syncing. Then the op-node is unable to see the engine sync complete, until the op-node restarts.